redis/client.py

```diff
@@ -572,63 +548,25 @@ def _disconnect_raise(self, conn, error):
         ):
             raise error
 
-    def _get_from_local_cache(self, command: str):
-        """
-        If the command is in the local cache, return the response
-        """
-        if (
-            self.client_cache is None
-            or command[0] in self.cache_blacklist
-            or command[0] not in self.cache_whitelist
-        ):
-            return None
-        while not self.connection._is_socket_empty():
-            self.connection.read_response(push_request=True)
-        return self.client_cache.get(command)
-
-    def _add_to_local_cache(
-        self, command: Tuple[str], response: ResponseT, keys: List[KeysT]
-    ):
-        """
-        Add the command and response to the local cache if the command
-        is allowed to be cached
-        """
-        if (
-            self.client_cache is not None
-            and (self.cache_blacklist == [] or command[0] not in self.cache_blacklist)
-            and (self.cache_whitelist == [] or command[0] in self.cache_whitelist)
-        ):
-            self.client_cache.set(command, response, keys)
-
-    def delete_from_local_cache(self, command: str):
-        """
-        Delete the command from the local cache
-        """
-        try:
-            self.client_cache.delete(command)
-        except AttributeError:
-            pass
-
     # COMMAND EXECUTION AND PROTOCOL PARSING
     def execute_command(self, *args, **options):
         """Execute a command and return a parsed response"""
         command_name = args[0]
         keys = options.pop("keys", None)
-        response_from_cache = self._get_from_local_cache(args)
+        pool = self.connection_pool
+        conn = self.connection or pool.get_connection(command_name, **options)
+        response_from_cache = conn._get_from_local_cache(args)
         if response_from_cache is not None:
             return response_from_cache
```

Comment:
Isn't this a connection leak if we pull from cache? 

We're getting a connection from the pool and never releasing it in this code path.

Reply:
Yes, it was, but fixed already in #3160